integrationTests/relayers/elrondToEth_test.go

```diff
@@ -1,251 +1,220 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRelayersShouldExecuteTransferFromElrondToEth(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	numTransactions := 2
+	deposits, tokensAddresses, erc20Map := createTransactions(numTransactions)
+
+	tokens, availableBalances := availableTokensMapToSlices(erc20Map)
+	safeContractEthAddress := testsCommon.CreateRandomEthereumAddress()
+	erc20ContractsHolder := createMockErc20ContractsHolder(tokens, safeContractEthAddress, availableBalances)
+
+	ethereumChainMock := mock.NewEthereumChainMock()
+	ethereumChainMock.SetQuorum(3)
+	expectedStatuses := []byte{bridge.Executed, bridge.Rejected}
+	ethereumChainMock.GetStatusesAfterExecutionHandler = func() []byte {
+		return expectedStatuses
+	}
+	elrondChainMock := mock.NewElrondChainMock()
+	for i := 0; i < len(deposits); i++ {
+		elrondChainMock.AddTokensPair(tokensAddresses[i], deposits[i].Ticker)
+	}
+	pendingBatch := mock.ElrondPendingBatch{
+		Nonce:          big.NewInt(1),
+		ElrondDeposits: deposits,
+	}
+
+	elrondChainMock.SetPendingBatch(&pendingBatch)
+
+	numRelayers := 3
+	relayers := make([]bridgeComponents, 0, numRelayers)
+	defer func() {
+		for _, r := range relayers {
+			_ = r.Close()
+		}
+	}()
+
+	messengers := integrationTests.CreateLinkedMessengers(numRelayers)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*1200)
+	defer cancel()
+	elrondChainMock.ProcessFinishedHandler = func() {
+		cancel()
+	}
+
+	for i := 0; i < numRelayers; i++ {
+		argsBridgeComponents := createMockBridgeComponentsArgs(i, messengers[i], elrondChainMock, ethereumChainMock)
+		argsBridgeComponents.Configs.GeneralConfig.Eth.SafeContractAddress = safeContractEthAddress.Hex()
+		argsBridgeComponents.Erc20ContractsHolder = erc20ContractsHolder
+		relayer, err := factory.NewEthElrondBridgeComponents(argsBridgeComponents)
+		require.Nil(t, err)
+
+		elrondChainMock.AddRelayer(relayer.ElrondRelayerAddress())
+		ethereumChainMock.AddRelayer(relayer.EthereumRelayerAddress())
+
+		go func() {
+			err = relayer.Start()
+			integrationTests.Log.LogIfError(err)
+			require.Nil(t, err)
+		}()
+
+		relayers = append(relayers, relayer)
+	}
+
+	<-ctx.Done()
+
+	transactions := elrondChainMock.GetAllSentTransactions(context.Background())
+	assert.Equal(t, 1, len(transactions))
+	assert.Nil(t, elrondChainMock.ProposedTransfer())
+	assert.Nil(t, elrondChainMock.PerformedActionID())
+
+	transfer := ethereumChainMock.GetLastProposedTransfer()
+	require.NotNil(t, transfer)
+
+	require.Equal(t, numTransactions, len(transfer.Amounts))
+
+	for i := 0; i < len(transfer.Amounts); i++ {
+		assert.Equal(t, deposits[i].To, transfer.Recipients[i])
+		assert.Equal(t, tokensAddresses[i], transfer.Tokens[i])
+		assert.Equal(t, deposits[i].Amount, transfer.Amounts[i])
+	}
+}
+
+func TestRelayersShouldExecuteTransferFromElrondToEthIfTransactionsAppearInBatch(t *testing.T) {
```

Comment:
perhaps 2 functions can be created here and reuse them in both tests.
eq:
- L154-L169, for loop is the same as in the first tests
- L108-L129 same with L25-L47
- L137-L151 same with L49-L63
- check are the same after ctx.Done()